### PR TITLE
Add integration example

### PR DESCRIPTION
**Pull request by KanaiYuma-aist**
aiaccel.torch.app.train と aiaccel.optimize.app.optimize を併用した、ブラックボックス最適化の example コードを作成しました

### CODE REVIEW

**Comment by KanaiYuma-aist** [comment]
jobs ベースの optimize に合わせて、改修を行いました。

- SaveValLossCallback を追加し、validation loss を out_filename に出力するように変更しました
- 不要ファイルを削除しました
- README を改修しました

**Review by yoshipon — CHANGES_REQUESTED** [review]

File: examples/integration/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+```bash
+pip install torchvision
+```
+
+
+Run the following command to perform black-box optimization.
+PATH_TO_ENV should be changed to the path of the environment prepared above.
+
+```bash
+python -m aiaccel.hpo.apps.optimize "python -m aiaccel.jobs.cli.abci3 gpu --command_prefix 'cd \$PBS_O_WORKDIR && module load cuda/12.6/12.6.1 && module load python/3.13/3.13.2 && source PATH_TO_ENV/bin/activate &&' jobs/{job_name}.log -- python -m aiaccel.torch.apps.train resnet50/config.yaml task.optimizer_config.optimizer_generator.lr={lr} trainer.logger.name=lr_{lr} out_filename={out_filename}" --config config.yaml
```
Comment (yoshipon) [inline examples/integration/README.md:18]:
```suggestion
aiaccel-hpo optimize --config hpo_config.yaml -- \
    aiaccel-job --config job_config.yaml train --n_gpus=1 jobs/{job_name}.log -- \
        aiaccel-torch train train_config.yaml \
            working_directory=jobs/{job_name}/ \
            task.optimizer_config.optimizer_generator.lr={lr} \
            out_filename={out_filename}
```
#478 に合わせて↑でいかがでしょうか．

Comment (yoshipon) [reply examples/integration/README.md:18]: マジですか！どんなエラーでましたでしょうか。
Slackなので伺えると幸いです。

Comment (KanaiYuma-aist) [reply examples/integration/README.md:18]: パスの件とは別件ですが、
working_directory=jobs/{job_name}/ \
の部分で、ディレクトリが存在しない旨のエラーが出ます。

機能的には aiaccel.torch.app.train に working_directory 作成の機能があるのが自然に思えたので、
その改修を別 pull request で行いました。

https://github.com/aistairc/aiaccel/pull/494

File: examples/integration/resnet50/config.yaml
Comment (yoshipon) [inline examples/integration/resnet50/config.yaml:1]: こちらですが，common_configをこちらにべた書きして頂いても良いでしょうか・
できるだけシンプルな方が良いと思います．

File: examples/integration/pyproject.toml
Comment (yoshipon) [inline examples/integration/pyproject.toml:1]: こちらもインストールに不要な行あれば削除お願いします．
パッケージ名は，中身を表す感じにできるとベターだと思いました．

File: examples/integration/task/torchvision_task_integration.py
Comment (yoshipon) [inline examples/integration/task/torchvision_task_integration.py:1]: こちらpackage_name/classification_task.py でお願いします．

File: examples/integration/README.md
Comment (yoshipon) [inline examples/integration/README.md:1]: こちらjobはデフォルトでlocalでまず動かしてもらう形は如何でしょう？
初手ABCIだとハードル高いなと思いまして．

**Review by yoshipon — CHANGES_REQUESTED** [review]
ありがとうございます．いくつかコメントしましたので，ご確認頂けると幸いです．
チュートリアル読む人が，コードを読んで理解するとき，できるだけ短い方がとっつきやすくて良いかと思いました．

File: examples/integration/task/torchvision_task_integration.py
Comment (KanaiYuma-aist) [reply examples/integration/task/torchvision_task_integration.py:1]: package_name は、これをそのままディレクトリ名にしますか？
また、examples/torch にも、該当ファイルと同機能のモジュールが存在しますが、こちらも classification_task.py に改名しますか？
（同じファイル名が複数あると、mypy でエラーが発生しますが・・・）

File: examples/integration/pyproject.toml
Comment (KanaiYuma-aist) [reply examples/integration/pyproject.toml:1]: 対応しました
4fb787d2efaef66473201e75431cac03a1fe8d3e

Comment (KanaiYuma-aist) [reply examples/integration/pyproject.toml:1]: 対応しました
パッケージ名は task_for_integration_example としました
4fb787d2efaef66473201e75431cac03a1fe8d3e
bb3a2266edf5fc1433fae61a7af21545aaf9aa41

**Comment by KanaiYuma-aist** [comment]
- qsub コマンドの job_group に関して、 `$JOB_GROUP` で環境変数を参照する方法だと hpo で subprocess を用いる箇所で環境変数が参照できなくなるので、job_config 内部に job_group を記載する方法に変更しました
- config_torch の `_base_ `指定に関して、resolve_pkg_path を用いるほうがスマートかもしれません
  - その場合、 examples/torch/image_classification に `__init__.py` が必要になります

**Review by yoshipon — APPROVED** [review]
ありがとうございます！
